django/contrib/admin/options.py

```diff
@@ -1343,6 +1346,44 @@ def response_delete(self, request, obj_display, obj_id):
                                current_app=self.admin_site.name)
         return HttpResponseRedirect(post_url)
 
+    def response_not_delete(self, request, obj_display, msg=''):
```

Comment:
I'm not convinced we need such a fancy solution -- after all, this is only for users who manually bypass the delete confirmation page.